### PR TITLE
Autocrafter groups support

### PR DESCRIPTION
**Pull request by SwissalpS**
Based on work started by @rubenwardy https://gitlab.com/rubenwardy/pipeworks/-/commit/60673db48058539915e2fa2ffa4fd3ef21f421fb

- adds group support to autocrafter

Set recipe as usual in formspec or via digilines. Now autocrafter will use up any items that conform to actual recipe that was registered with groups.

e.g. set recipe to:
- any trunk, any leaves, any leaves
- any leaves, any leaves, any leaves
- any leaves, any leaves, any leaves
-> result output is set to bonemeal:mulch

- now fill input inventory with various trunks not used in recipe and leaves not used when setting recipe
- start autocrafter and watch it use up the available items

closes: #3

Edit: also snuck in some whitespace edits 9be6239

### CODE REVIEW

File: autocrafter.lua [hunk 1]
```diff
@@ -21,58 +22,190 @@ local function get_item_info(stack)
...
+		score = 0
+		index_recipe = count_index(recipes[i].items)
+		for recipe_item_name, _ in pairs(index_recipe) do
+			if index_example[recipe_item_name] then
+				score = score + 1
+			elseif recipe_item_name:sub(1, 6) == "group:" then
+				group = recipe_item_name:sub(7)
+				for example_item_name, _ in pairs(index_example) do
+					def = minetest.registered_items[example_item_name]
+					if def and def.groups[group] and def.groups[group] > 0 then
```
Comment (fluxionary) [inline autocrafter.lua:50]: this could be replaced w/
```lua
minetest.get_item_group(example_item_name, group) > 0
```

Comment (SwissalpS) [reply autocrafter.lua:50]: thanks, will do.

**Comment by wsor4035** [comment]
has conflicts

**Comment by BuckarooBanzay** [comment]
> Now I'm in the middle of relocating again, to an off-the-grid place. It will take a while until I have power, internet and time to rebase this.

Good luck 👍 don't pressure yourself, this can wait 😄

**Comment by dontfwiththestream** [comment]
> > has conflicts
> 
> I'll take a look at this once I have glasses again.
> 
> (Although, it's been sitting here quietly for months. Not sure if there is any interest for this to be solved :p)

There is at least from my end!

**Comment by wsor4035** [comment]
conflicts should be resolved ™️

**Comment by BuckarooBanzay** [comment]
bump! this _looks_ finished :thinking:  could this potentially be merged soon?

**Review by wsor4035 — APPROVED** [review]
seems legit

**Comment by OgelGames** [comment]
I don't know if this was ready for merging... hopefully it doesn't effect performance too much, checking every item in the inventory for each craft doesn't seem very efficient.

**Pull request merged**